leftwm-core/src/handlers/focus_handler.rs

```diff
@@ -152,12 +158,22 @@ impl State {
 }
 
 fn focus_workspace_work(state: &mut State, workspace_id: Option<i32>) -> Option<()> {
-    //no new history if no change
+    // no new history if no change
     if let Some(fws) = state.focus_manager.workspace(&state.workspaces) {
         if fws.id == workspace_id {
             return None;
         }
     }
+
+    // Add the currently focused window to the history
+    if let Some((tag, handle)) = state
+        .focus_manager
+        .tag(0)
+        .zip(state.focus_manager.window(&state.windows).map(|w| w.handle))
+    {
+        state.focus_manager.tags_last_window.insert(tag, handle);
+    }
+
```

Comment:
I think this being in focus_tag_work would make more sense.

Reply:
humm, something's broken if I do that, `FocusWorkspaceNext` works, but `FocusWorkspacePrevious` somehow focus the top window on the current workspace.

Reply:
hum, this isn't actually related to this, it's because the cursor got moved to the center of the screen, which was only showing the margin between my 2 windows, so the focus got borked until I moved the cursor.